distribution/docker/Dockerfile

```diff
@@ -51,12 +52,19 @@ RUN --mount=type=cache,target=/root/.m2 VERSION=$(mvn -B -q org.apache.maven.plu
 
 FROM busybox:1.34.1-glibc as busybox
 
-FROM gcr.io/distroless/java$JDK_VERSION-debian11
+FROM $BASE_IMAGE
+ARG JDK_VERSION
+ARG BASE_IMAGE
+
 LABEL maintainer="Apache Druid Developers <dev@druid.apache.org>"
 
 COPY --from=busybox /bin/busybox /busybox/busybox
-RUN ["/busybox/busybox", "--install", "/bin"]
-
+RUN ["/busybox/busybox", "sh", "-c", "if [ $BASE_IMAGE = gcr.io/distroless/java$JDK_VERSION-debian11 ]; then \
```

Comment:
instead of comparing the base image I would suggest we simply test if we can run `bash` in which case we can assume we have a base image that doesn't require installing busybox or bash to begin with